scale/batch/definition/definition.py

```diff
@@ -49,8 +50,19 @@ def validate(self, batch):
                 self.prev_batch_diff.set_force_reprocess(self.forced_nodes)
             if not self.prev_batch_diff.can_be_reprocessed:
                 raise InvalidDefinition('PREV_BATCH_NO_REPROCESS', 'Previous batch cannot be reprocessed')
-        # New batch
         
+        # New batch - need to validate dataset parameters against recipe revision
+        elif self.dataset:
+            from data.models import DataSet
+            from recipe.models import RecipeType
+            dataset = DataSet.objects.get(pk=self.dataset)
+            dataset_definition = dataset.get_definition()
+            recipe_type = RecipeType.objects.get(name=batch.recipe_type.name, revision_num=batch.recipe_type_rev.revision_num)
```

Comment:
Use RecipeTypeRevision.objects.get_revision() method and get the definition from the revision. The recipetype get method will fail here if the revision number requested isn't the latest version.  The RecipeTypeRevision class stores the definition for the revision and the RecipeType class stores the elements that are common to each revision.